src/lizaalert/courses/models.py

```diff
@@ -418,6 +418,53 @@ class Meta:
         ordering = ("knowledge",)
 
 
+class Cohort(TimeStampedModel):
+    """
+    Модель для представления группы курса.
+
+    Поля модели:
+    - course: ForeignKey к модели Course
+    - cohort_number: уникальный номер группы в рамках курса
+    - start_date: дата начала обучения
+    - end_date: дата окончания обучения
+    - students_count: количество студентов в группе
+    - teacher: имя преподавателя группы
+    """
+
+    course = models.ForeignKey(Course, on_delete=models.PROTECT, related_name="cohorts", verbose_name="Курс")
+    cohort_number = models.PositiveIntegerField(verbose_name="Номер группы")
+    start_date = models.DateField(verbose_name="Дата начала", null=True, blank=True)
+    end_date = models.DateField(verbose_name="Дата окончания", null=True, blank=True)
+    students_count = models.PositiveIntegerField(verbose_name="Количество студентов", null=True, blank=True)
+    teacher = models.ForeignKey(User, on_delete=models.PROTECT, verbose_name="Преподаватель")
+
+    def save(self, *args, **kwargs):
+        if not self.pk:
+            last_cohort = Cohort.objects.filter(course=self.course).order_by("-cohort_number").first()
```

Comment:
```suggestion
            last_cohort = Cohort.objects.filter(course=self.course).aggregate(max_number=Max('cohort_number'))['max_number']
```
Нам не нужен весь объект последней когорты, нам необходим только самый большой порядковый номер, его проще высчитать и получить на стороне БД.